game/qsp.ksy

```diff
@@ -0,0 +1,353 @@
+meta:
+  id: qsp
+  title: Quest Soft Player (QSP) Game
+  application: Quest Soft Player
+  file-extension: qsp
+  endian: le
+  encoding: "utf-16le" # in fact "ucs-2le"
+  license: Unlicense
+doc: |
+  QSP is a player for text quests written in a QSP language.
+
+doc-ref:
+  - https://github.com/QSPFoundation/qsp/blob/master/help/gam_desc.txt
+
+seq:
+  - id: signature
+    contents: ['Q',0, 'S',0, 'P',0, 'G',0, 'A',0, 'M',0, 'E',0, 0x0D,0, 0x0A,0] #QSPGAME
+    if: is_new_format
+  - id: version
+    type: usual_str
+    if:  is_new_format
+  - id: password
+    type: rot5_str
+    if:  is_new_format
+  - id: locations_count_str
+    type:
+      switch-on: is_new_format
+      cases:
+        true: rot5_number_str
+        false: number_str
+  - id: locations
+    type: loc
+    repeat: expr
+    repeat-expr: locations_count
+instances:
+  is_new_format:
+    value: true
+  locations_count:
+    value: "(is_new_format?locations_count_str.as<rot5_number_str>.number:locations_count_str.as<number_str>.number)"
+types:
+  string_placeholder:
```

Comment:
Ok, so this is a very hacky implementation. Looks like what it's doing is essentially scanning for [0xd, 0xa] in 2-byte UTF16 character units. Likely will benefit from implementation of https://github.com/kaitai-io/kaitai_struct/issues/538

Reply:
Added the ref to the issue.